csrc/scheduler/hopper_multi_matmul.cpp

```diff
@@ -434,60 +434,52 @@ void HopperMultipleMatmulScheduler::scheduleMmaResults() {
   }
 }
 
-void HopperMultipleMatmulScheduler::scheduleOutputTensor(TensorView* c) {
-  // Block Schedule and Parallelize
-  blockTileTensors({c});
-  parallelizeBlocks({c});
-
-  // Apply mma common transformation
-  c->split(-2, getM(params_->mma_macro));
-  c->split(-1, getN(params_->mma_macro));
-  // [..., Mo, No, Mio, Mii, Nio, Nii]
-  // -> [..., Mo, No, Mio, Nio, Mii, Nii]
-  c->reorder({{-3, -2}});
-  c->merge(-4);
-
-  // [...., Mii, Nii] ->
-  // [...,  Mii/16,  Miioi(2), Miii(8), Nii/8, Niio(4), Niii(2)] ->
-  // [.., Mii/16,  Miii(8), Niio(4),  Nii/8, Miioi(2), Niii(2)   ]
-  // [..., Mii/16 * Miii(8) * Niio(4), Nii/8, Miioi(2), Niii(2)  ]
-  // Mii/16 * Miii(8) * Niio(4) is 128 for Hopper and this is parallelized as
-  // TIDx.
-  auto s =
-      mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(c->getLoopDomain());
-  c->setLoopDomain(s.as<IterDomain*>());
-  c->axis(-5)->parallelize(ParallelType::TIDy);
-}
-
 void HopperMultipleMatmulScheduler::scheduleEpilogue() {
+  // Load/cache the epilogue inputs if there are any.
+  auto& c_tvs = tensor_roles_.at(MatmulTensorRole::EPILOGUE_INPUT);
+  std::vector<TensorView*> cached_tvs;
+  for (auto* c : c_tvs) {
+    cached_tvs.push_back(c->cacheAfter());
+  }
+
+  // Propato to (not including) the splitk output if there is a splitk
+  // else this is just mma_results_
+  std::vector<TensorView*> propagate_to = splitk_sums_;
+  if (tensor_roles_.count(MatmulTensorRole::EPILOGUE_INPUT)) {
+    propagate_to.insert(propagate_to.end(), c_tvs.begin(), c_tvs.end());
+  }
+
   if (!params_->use_smem_epilogue) {
     for (Val* dv : fusion_->outputs()) {
       auto* d = dv->as<TensorView>();
       NVF_ERROR(d->definition() && d->definition()->isA<LoadStoreOp>());
 
       // Schedule the output TV and propagate it back to the outputs of the Mma
       // op.
-      scheduleOutputTensor(d);
+      blockTileTensors({d});
+      parallelizeBlocks({d});
+      transformLikeMmaOutput(d, /*is_mma_result=*/false);
+
+      auto s = mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(
+          d->getLoopDomain());
+      d->setLoopDomain(s.as<IterDomain*>());
+
       scheduler_utils::BoundedDirectionalTransformPropagator::backward(
           d,
           -1,
-          mma_results_,
+          propagate_to,
           scheduler_utils::BoundedDirectionalTransformPropagator::Options()
```

Comment:
Just a note: we'll need to revisit this path for cases where TMA is disabled. We need to make sure it is doing coalesced stores here (can check with ncu). I imagine that we will almost always want to use TMA, but it's possible that we would skip it if it increases smem usage too much. In those cases we'll hit this path.